core/packages/contracts/contracts/XcmApp.sol

```diff
@@ -0,0 +1,145 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.9;
+
+import "./ChannelRegistry.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+// TODO: put encoding functions here
+//import "./XCMAppPallet.sol";
+
+/// @title Proxy
+/// @notice A simple pass through proxy.
+contract Proxy is Ownable {
```

Comment:
should we call this SovereignAccount instead of Proxy?

Reply:
Nope, I think Proxies will eventually be used for all kinds of accounts on Polkadot. Specifically, `DescendOrigin` will result in another layering of proxying for simple accounts on parachains.

Reply:
Maybe simple accounts are still SovereignAccounts though? We should confirm if there is even a distinction between the two

Reply:
I don't think so. My reading of the xcm spec leads me to believe that only _consensus systems_ can have sovereign accounts, where a consensus system is a relaychain, parachain, pallet, or a contract within a pallet. Simple accounts are not consensus systems because they don't have specialized consensus code. They are more like data structures.

Indeed the xcm spec refers to simple accounts as _Locations_:

>Location: A Consensus System, or an addressable account or data structure that exists therein. Examples include the Treasury account on the Polkadot Relay-chain

Reply:
ok. though in the ethereum context there is no difference really